cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -1903,50 +1920,135 @@ pub(super) mod test {
         }
     }
 
-    #[test]
-    fn infers_id_from_github_flake_ref() {
-        let url = Url::parse("github:foo/bar").unwrap();
-        eprintln!("{:?}", url);
-        let inferred = infer_flake_install_id(&url).unwrap();
-        assert_eq!(inferred.as_str(), "bar");
-    }
-
-    #[test]
-    fn infers_id_from_https_flake_ref() {
-        let url = Url::parse("https://github.com/foo/bar/archive/main.tar.gz").unwrap();
-        let inferred = infer_flake_install_id(&url).unwrap();
-        assert_eq!(inferred.as_str(), "main.tar.gz");
-    }
-
-    #[test]
-    fn infers_fallback_id_for_https_flake_ref() {
-        let url = Url::parse("https://example.com/foo/bar/baz").unwrap();
-        let inferred = infer_flake_install_id(&url).unwrap();
-        assert_eq!(inferred.as_str(), "baz");
+    /// Determines whether to have a branch and/or revision in the URL
+    #[derive(Debug, Arbitrary, PartialEq)]
+    enum FlakeRefPathAttrs {
+        None,
+        RevPath,
+        RevParam,
+        RefParam,
+        RefAndRevParams,
+    }
+
+    /// The components of an attrpath after `packages.<system>.`
+    #[derive(Debug, Arbitrary, PartialEq)]
+    enum AttrPathComponent {
+        Bare,
+        Quoted,
+        QuotedWithDots,
+    }
+
+    /// The type of URL in the flake reference
+    #[derive(Debug, Arbitrary, PartialEq)]
+    enum FlakeRefURLType {
+        GitHub,
+        Https,
+        GitHttps,
+    }
+
+    /// Flake ref outputs
+    #[derive(Debug, Arbitrary, PartialEq)]
+    enum FlakeRefOutputs {
+        None,
+        All,
+        Out,
+        OutAndMan,
+    }
+
+    #[derive(Debug, Arbitrary, PartialEq)]
+    enum PkgFragment {
+        None,
+        #[proptest(
+            strategy = "proptest::collection::vec(any::<AttrPathComponent>(), 1..=2).prop_map(PkgFragment::AttrPath)"
+        )]
+        AttrPath(Vec<AttrPathComponent>),
+    }
+
+    #[derive(Debug, Arbitrary)]
+    struct ArbitraryFlakeRefURL {
+        url_type: FlakeRefURLType,
+        path_attrs: FlakeRefPathAttrs,
+        pkg_fragment: PkgFragment,
+        outputs: FlakeRefOutputs,
+    }
+
+    fn arbitrary_flake_ref_url() -> impl Strategy<Value = (String, String)> {
```

Comment:
note nonblocking: in general I think adding proptest stuff is great, but I think we need to be wary of
1. making tests hard to read
and
2. having logic in our tests

Not sure if I agree with it or how trustworthy it is, but here's a random blog post as food for thought: https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

For this function, I would want all of the logic generating `expected_install_id` commented and probably commented with cases we care about. I might even want to keep all those cases as unit tests, because they're so fast. And they're way easier to review. E.g. you made a change for when `last()` doesn't make sense for github flakerefs. If I can just see you didn't regress any tests, and you added a case covering something I can glance at and see makes sense, that's way easier than trying to figure out if and how it's covered by proptest.